test/test-fs.c

```diff
@@ -1271,6 +1271,7 @@ TEST_IMPL(fs_mkstemp) {
   int r;
   int fd;
   const char path_template[] = "test_file_XXXXXX";
+  const char test_file[] = "test_file";
```

Comment:
```suggestion
  char test_file[] = "test_file";
```
A `const char` array is immutable and may be in read-only memory.

Reply:
That is the goal, no? The parameter to the function is marked `const char* tpl`, so we should not be mutating it.

Reply:
Okay, I guess I don't understand why this is hoisted into a variable then, if it's a semantically null change.

Reply:
@bnoordhuis The array needs to be mutable, since it will be replaced by null string on error. Old test case will fail because the string is in read-only area. Using `const char` array, I found that a copy is created in stack for my machine and same is the case with all the target architectures tested using github CI (otherwise there should be segmentation fault) but i'm not sure it's the standard or not.
Should i keep `const char` or move to `char` ?

Reply:
Right, I understand what Jameson means now. Can you undo this change? It's just confusing.

`req->path` is a copy of the `path` argument to `uv_fs_mkstemp()`, i.e., `path` is never mutated and can just be a string literal, like it was before.

Reply:
I missed  the strdup part in `uv_fs_mkstemp`. Pushed the suggested changes